### DISCUSSION TITLE
Aggiunta di una tipologia di voto a quelle disponibili

### DISCUSSION THREAD

**Discussion by Lucaccc** [iisgiua/giuaschool#641] [Richieste di aiuto] [answered]
Buongiorno, volevo aggiungere una tipologia di voto (ad esempio la giustificazione) tra quelle disponibili già nel menù a tendina. 
Potresti indicarmi in quali file devo andare ad attuare le modifiche per poterlo fare? Non credo sia solo una questione di template, per cui vorrei sapere in quali file devo andare a "mettere le mani". 
Grazie mille.

**Comment by trinko** [comment]
Non mi è chiaro cosa intendi fare: se parli di menu a tendina, immagino tu stia pensando alla pagina **SCUOLA > MATERIA**, ma questa serve per il tipo di valutazione usata per gli scrutini. Non ha influenza sull'inserimento dei voti nelle lezioni.

Se vuoi solo aggiungere simboli (+/-/i ecc..) potresti farlo usando il campo del _giudizio_ presente per il voto. Puoi modificare i template in modo da visualizzare diversamente il campo giudizio quando corrisponde al simbolo desiderato. E aggiungere pulsanti per l'inserimento rapido del simbolo (con un po' di Javascript).
In questo caso è sufficiente modificare il template, perchè per la memorizzazione sul database non cambia nulla.

I template della gestione voti per i docenti sono:
- **templates/lezioni/voti_alunno_edit.html.twig**: pagina di inserimento voto per un alunno
- **templates/lezioni/voti_classe_edit.html.twig**: pagina di inserimento voti per la classe
- **templates/lezioni/voti_dettagli.html.twig**: pagina di dettaglio dei voti di un alunno
- **templates/lezioni/voti_quadro.html.twig**: pagina con il quadro generale dei voti

Mentre per i genitori/alunni:
- **templates/ruolo_genitore/voti.html.twig**: pagina con il quadro dei voti
- **templates/ruolo_genitore/voti_materia.html.twig**: pagina con il dettaglio dei voti per una materia

**Reply by Lucaccc** [reply]
Se volessi utilizzare il campo giudizio come dovrei fare? Dove devo inserire il codice e cosa devo scriverci?

**Comment by trinko** [comment]
> Se volessi utilizzare il campo giudizio come dovrei fare? Dove devo inserire il codice e cosa devo scriverci?

Nel template di inserimento dei voti puoi aggiungere un pulsante e con un po' di codice Javascript (o JQuery) fai in modo che quando si clicca sul pulsante si inserisca in automatico il simbolo che hai deciso nel campo del form del giudizio, inviando poi i dati alla pagina (submit).

Nei template di visualizzazione dei voti, invece, puoi usare il comando del template _{% if ... %}...{% else %}...{% endif %}_ per visualizzare, quando nel giudizio si trova il simbolo che hai deciso di usare, quello che vuoi al posto del normale asterisco.

**Reply by Davide031** [reply]
Se volessi aggiungere il voto +o- con valore 6.00 e 4.00 come posso fare?

**Comment by Lucaccc** [comment]
> > Se volessi utilizzare il campo giudizio come dovrei fare? Dove devo inserire il codice e cosa devo scriverci?
> 
> Nel template di inserimento dei voti puoi aggiungere un pulsante e con un po' di codice Javascript (o JQuery) fai in modo che quando si clicca sul pulsante si inserisca in automatico il simbolo che hai deciso nel campo del form del giudizio, inviando poi i dati alla pagina (submit).
> 
> Nei template di visualizzazione dei voti, invece, puoi usare il comando del template _{% if ... %}...{% else %}...{% endif %}_ per visualizzare, quando nel giudizio si trova il simbolo che hai deciso di usare, quello che vuoi al posto del normale asterisco.

Buongiorno.
Nei template indicati non trovo il codice che fa vedere l'asterisco quando la valutazione non è numerica, ma è presente solo il giudizio...Dove posso trovarlo?
Grazie.

**Comment by trinko** [comment]
> Buongiorno. Nei template indicati non trovo il codice che fa vedere l'asterisco quando la valutazione non è numerica, ma è presente solo il giudizio...Dove posso trovarlo? Grazie.

Ad esempio nella riga 133 del template **templates/lezioni/voti_quadro.html.twig** o nella riga 49 di **templates/ruolo_genitore/voti.html.twig**:
```
<button type="button" class="btn btn-xs btn-default" title="Informazioni sulla valutazione" data-toggle="modal" data-target="#gs-modal-info" data-title="Informazioni sulla valutazione" data-body="gs-voti-{{ v.id }}"><span class="glyphicon glyphicon-asterisk"></span></button>

```

**Reply by Lucaccc** [reply]
Buongiorno.
In questo modo posso solo cambiare l'asterisco con il simbolo che voglio vedere e sono riuscito...come faccio, invece, a inserire un comando del tipo "se nel giudizio è presente il simbolo x, allora mostra questo simbolo"?
Grazie mille.

**Reply by trinko** [reply, marked as answer]
> come faccio, invece, a inserire un comando del tipo "se nel giudizio è presente il simbolo x, allora mostra questo simbolo"? Grazie mille.

Devi usare il comando _if_, ad esempio:
```
{% if v.giudizio == 'x' %}qualcosa{% else %}altro{% endif %} 
```

**Reply by Lucaccc** [reply]
Grazie mille, sono riuscito.

**Comment by ghost** [comment]
@Lucaccc Buongiorno! Avrei bisogno di parlarti, in merito a questa richiesta. Posso parlarti in privato? Mi scrivi alla mia mail: padovesi.roberto@gmail.com? Grazie!

**Comment by robertopadovesi** [comment]
Ciao, @Lucaccc! Avrei una domanda da farti, per caso, conosci una certa Federica Bianchi, che ti abbia chiesto di qualche scuola a cui potersi rivolgersi? Non è una domanda spam o una domanda phising, ma è solo un informazione. Grazie!
